### ISSUE TITLE
The question about training stage in faster_rcnn

### ISSUE THREAD

**Issue by TIEHua** [ZiweiWangTHU/BiDet#23]
Hello, I'm sorry to bother you. I met some problems in the training of BIDet's Fasters-RCNN, and I would like to consult you:

During training, in the first few epochs, there will be a loss NaN on some Iteration sections;

My training orders are: **Python faster_rCNN /trainval_net.py --dataset='coco' --data_root='data/coco' --basenet='pretrain/resnet18.pth' -- mGPUs=True** 
I set RPN_PRIOR_WEIGHT, RPN_REG_WEIGHT, HEAD_PRIOR_WEIGHT, AND HEAD_PRIOR_WEIGHT as 0.2, 0.1, 0.2, 0.1 respectively from the [begining.]

Have you ever encountered this problem in the training process?

**Comment by Wuziyi616** [comment]
Yeah I have also encountered such problems. I think the reason is that, training BNN on large networks like Faster R-CNN is very unstable, so sometimes the loss explodes. And applying BiDet makes it more unstable. My suggestion is to first train without BiDet loss (that is to say, set RPN_PRIOR_WEIGHT, RPN_REG_WEIGHT, HEAD_PRIOR_WEIGHT, AND HEAD_PRIOR_WEIGHT to 0) before the first learning rate decay epoch, and then add these losses to improve the performance.

**Comment by TIEHua** [comment]
Thanks for your reply. I've tried to do that. But I noticed that the RPN_PRIOR_WEIGHT, RPN_REG_WEIGHT, HEAD_PRIOR_WEIGHT, AND HEAD_PRIOR_WEIGHT are set to 0.01, 0.5, 0.05, 0.02 in the 7st epoch (after the first learning rate decay). 
In the **class fasterRCNN_Bidet**, you set these parameters to 0.2 0.1 0.2 0.1.  Which parameters should I use？
In fact, I found that using 0.2,0.1,0.2,0.1 still led to loss explosions.

**Comment by Wuziyi616** [comment]
I think I used 0.01, 0.5, 0.05, 0.02, while the default parameters in the __init__ function is not the optimal one we use. It's really strange that the loss explodes after the first lr decay. I check the training log for Faster R-CNN yesterday and found it more stable than SSD, mainly because ResNet backbone is more 'robust' than VGG. Did you use the pre-trained backbone? What's your init lr?

**Comment by TIEHua** [comment]
> I think I used 0.01, 0.5, 0.05, 0.02, while the default parameters in the **init** function is not the optimal one we use. It's really strange that the loss explodes after the first lr decay. I check the training log for Faster R-CNN yesterday and found it more stable than SSD, mainly because ResNet backbone is more 'robust' than VGG. Did you use the pre-trained backbone? What's your init lr?

Thanks for your reply. I used the pre-trained model and init lr is 1e-3. When using 0.01, 0.5, 0.05, 0.02, the loss is fine. But when set 0.2, 0.1, 0.2, 0.1 head_prior_loss is easy to explore and make training unstable.
